### DISCUSSION TITLE
Fontconfig error: Cannot load default config file

### DISCUSSION THREAD

**Discussion by nodecentral** [ImageMagick/ImageMagick#4200] [Help]
### Description

Hi, 

Whatever I try to do I receive the following errors, everything still seems to work fine, but I’d like to address the errors if possible .

````
Fontconfig error: Cannot load default config file
Fontconfig error: Cannot load default config file
montage: unable to read font `helvetica' @ error/annotate.c/RenderFreetype/1432.
````

### Steps to Reproduce

````
bash -c 'convert cabin.jpg -resize 246x246\! a_e.jpg
convert conservatory.jpg -resize 246x246\! b_e.jpg
convert kitchen.jpg -resize 246x246\! c_e.jpg
convert frontroom.jpg -resize 246x246\! d_e.jpg
montage a_e.jpg b_e.jpg c_e.jpg d_e.jpg -geometry 246x246+2+2 collage.jpg'
````
or 
````
montage conservatory.jpg cabin.jpg frontroom.jpg kitchen.jpg -tile 2x2 assembly.jp
````

### System Configuration
Installed on a QNAP NAS 

Version: ImageMagick 7.0.7-28 Q16 x86_64 2018-05-14 http://www.imagemagick.org
Copyright: © 1999-2018 ImageMagick Studio LLC
License: http://www.imagemagick.org/script/license.php
Features: Cipher DPC HDRI OpenMP 
Delegates (built-in): bzlib djvu fftw fontconfig freetype gvc jbig jng jp2 jpeg lcms lqr lzma openexr pangocairo png raw tiff webp wmf xml zlib

**Comment by urban-warrior** [comment]
Add -font to your command line followed by a TrueType font file such as arial.ttf (e.g. -font arial.ttf).  That should resolve the issue.  Better yet, download and install the latest release of ImageMagick, 7.0.10-46.  It includes better font handling than 7.0.7-28.

**Reply by idallen** [reply]
AppImage version `ImageMagick 7.1.0-25 Q16-HDRI x86_64 2022-02-12` is not yet clever enough about fonts.
~~~~~
$ Downloads/magick montage ian.jpg some.jpg
montage: unable to read font `helvetica' @ error/annotate.c/RenderFreetype/1474.
~~~~~
I can get rid of the error using `-font courier` or using most any font from `xlsfonts` but the default gives the above error.
Ubuntu standard installed version `ImageMagick 6.9.10-23 Q16 x86_64 20190101` does not have this error.
Ubuntu 20.04.4 LTS

**Comment by nodecentral** [comment]
Many thanks, I’ll give it a try - QQ: do you know why it gives a font error message , even when it looks like I’m not doing anything ‘font’ related ? Sorry if that’s a stupid question, just curious ..

**Comment by nodecentral** [comment]
Hi

Just to let you know I tried your suggestion,  but sadly I still get an error message, this time showing the aerial font.

````
[/share/Public] # montage conservatory.jpg kitchen.jpg frontroom.jpg cabin.jpg -font arial.ttf  montage.jpg
Fontconfig error: Cannot load default config file
Fontconfig error: Cannot load default config file
montage: unable to read font `arial.ttf' @ warning/annotate.c/RenderType/1009.
montage: unable to read font `arial.ttf' @ error/annotate.c/RenderFreetype/1432.
````

Is there a font.conf file or something like that somewhere that I can find/edit ?

**Comment by urban-warrior** [comment]
You must download arial.ttf from the internet (use Google search) and specify the path to the file.  If its in your current directory, -font arial.ttf works.  Fontconfig is a separate delegate library that ImageMagick leverages.  Search Google for how to properly configure your Fontconfig subsystem to get rid of the 'Cannot load default...' warning.
